src/fty_common_db_asset.cc

```diff
@@ -718,6 +750,53 @@ select_assets_all_container (tntdb::Connection &conn,
     }
 }
 
+int
+select_asset_element_by_dc
+    (tntdb::Connection& conn,
+     int64_t dc_id,
+     std::function<void(const tntdb::Row&)>& cb)
+{
+    LOG_START;
+
+    try{
+        // use nested SELECT instead of three-table JOIN
+        std::string st =
+            "SELECT "
+            "   v.id, v.name, v.type_name, "
+            "   v.subtype_name, v.id_parent, "
+            "   v.status, v.priority, "
+            "   v.asset_tag "
+            " FROM "
+            "   v_web_element v "
+            "WHERE "
+            "   v.id in "
+            "   ( "
+                " SELECT p.id_asset_element "
+                " FROM v_bios_asset_element_super_parent p "
+                " WHERE "
+                "   :containerid in ( p.id_parent1, p.id_parent2, p.id_parent3, "
+                "                     p.id_parent4, p.id_parent5, p.id_parent6, "
+                "                     p.id_parent7, p.id_parent8, p.id_parent9, "
+                "                     p.id_parent10) "
+            "   ) ";
```

Comment:
Consider switching to C++ raw string literals.

Reply:
When using raw string literals, tntdb becomes implacably unhappy about the indentation, and removing indentation worsens the readability more than current approach.